### ISSUE TITLE
`Style/MultilineIfModifier` conflicts to `Style/GuardClause`

### ISSUE THREAD

**Issue by nepalez** [rubocop-hq/rubocop#3709]
The first example breaks `Style/MultilineIfModifier`

```ruby
if foo
  bar.baz
     .qux
end
```

The second one breaks `Style/GuardClause`

```ruby
bar.baz
   .qux if foo
```

**Comment by mikegee** [comment]
I can't get the first example to return an offense. Please post your Rubocop config.

**Comment by jaronson** [comment]
Well, well, well, well. Got Mr. @mikegee over here looking to be able to reproduce an issue.

Here's what I can see.

Given this code:

``` ruby
def upload_owner_image
  image = owner_params.delete(:image)
  if image.present?
    owner_params[:image_path] = upload_image(
      :owner,
      image
    )
  end
end
```

This is the complaint:

```
lib/resources/owners.rb:36:11: C: Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.
          if image.present?
```

Corrected to:

``` ruby
def upload_owner_image
  image = owner_params.delete(:image)
  owner_params[:image_path] = upload_image(
    :owner,
    image
  ) if image.present?
end
```

Complains with:

```
lib/resources/owners.rb:36:11: C: Style/MultilineIfModifier: Favor a normal if-statement over a modifier clause in a multiline statement.
          owner_params[:image_path] = upload_image( ...
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Ad infinitum.

Here's my cops.

```
AllCops:
  DisplayCopNames: true
  TargetRubyVersion: 2.3

Documentation:
  Enabled: false

Metrics/ClassLength:
  Max: 360

Metrics/ModuleLength:
  Max: 360

Metrics/MethodLength:
  Max: 30

Metrics/AbcSize:
  Max: 30

Metrics/BlockLength:
  Enabled: false

Style/AccessorMethodName:
  Enabled: false

Style/AlignHash:
  EnforcedColonStyle: table

Style/MultilineOperationIndentation:
  EnforcedStyle: indented

Style/PredicateName:
  Enabled: false

Style/SpaceAroundOperators:
  Enabled: false

Style/TrivialAccessors:
  IgnoreClassMethods: true

Style/MutableConstant:
  Enabled: true

Style/MultilineMethodCallIndentation:
  EnforcedStyle: indented

Style/SignalException:
  EnforcedStyle: only_fail

Style/IndentArray:
  EnforcedStyle: consistent
```

**Comment by mikegee** [comment]
Hi there Josh!

I can confirm the reported behavior even with the default config.

Can you offer a potential solution? I'm not sure what can be done about it other than disabling one cop or the other with an inline comment.

The examples given here can be made a single line, but I suppose they aren't realistic in some way. Can you confirm this conflict is problematic beyond the trivial examples posted so far?

**Comment by jaronson** [comment]
They can't be made a single line, though, if `Style/ClassAndModuleChildren: Use nested module/class definitions` is enforced and it's within a few nested modules and `Metrics/LineLength` is set to 80. This is the reason this error has cropped up for me.

Either of these cops could check the config to see if the other is enabled and skip itself.

A more complex solution might involve checking the value of `Metrics/LineLength` and guessing whether the break is happening for that reason and skipping itself. That sounds like it could be problematic though.

Maybe we just shouldn't use both styles.

**Comment by jaronson** [comment]
Or maybe an option on `Style/GuardClause` to skip multi lines?

**Comment by madwort** [comment]
@jaronson you could manually correct this:

```
def upload_owner_image
  image = owner_params.delete(:image)
  if image.present?
    owner_params[:image_path] = upload_image(
      :owner,
      image
    )
  end
end
```

to this:

```
def upload_owner_image
  image = owner_params.delete(:image)
  return unless image.present?
  owner_params[:image_path] = upload_image(
    :owner
    image
  )
end
```

**Comment by mijoharas** [comment]
Hi, I think `Style/GuardClause` should not apply to multiline statements. 

Also, I think this issue should be marked as a bug since there are two contradicting style recommendations. (I was trying to see if someone else had reported it by looking at the issues labelled bug and couldn't find it).

**Comment by madwort** [comment]
Hi @mijoharas , there are not two contradicting style recommendations. In @jaronson 's comment above https://github.com/bbatsov/rubocop/issues/3709#issuecomment-264246545 when he hits the GuardClause cop, he mistakenly uses a multiline if modifier instead of using a guard clause as suggested by the cop.

**Comment by mijoharas** [comment]
Sorry, thanks @madwort, upon rereading you're right.
